phoenix-core/src/test/java/org/apache/phoenix/tool/ParameterizedPhoenixCanaryToolIT.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.tool;
+
+import com.google.gson.Gson;
+import org.apache.phoenix.end2end.NeedsOwnMiniClusterTest;
+import org.apache.phoenix.end2end.ParallelStatsDisabledIT;
+import org.apache.phoenix.query.QueryServices;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+
+import static org.apache.phoenix.tool.PhoenixCanaryTool.propFileName;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+@Category(NeedsOwnMiniClusterTest.class)
+public class ParameterizedPhoenixCanaryToolIT extends ParallelStatsDisabledIT {
+
+	private static String connString = "";
+	private static Properties canaryProp = new Properties();
+	private static Connection connection = null;
+	private static final String stdOutSink = "org.apache.phoenix.tool.PhoenixCanaryTool$StdOutSink";
+	private static final String fileOutSink = "org.apache.phoenix.tool.PhoenixCanaryTool$FileOutSink";
+	private boolean isNamespaceEnabled;
+	private boolean isPositiveTestType;
+	private List<String> cmd = new ArrayList<>();
+	private String resultSinkOption;
+	private ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+
+	@Parameterized.Parameters(name="ParameterizedPhoenixCanaryToolIT_assertTrue={0},isNamespaceEnabled={1},resultSinkOption={2}")
+	public static Collection isNamespaceEnabled() {
+		return Arrays.asList(new Object[][] {
+						{true, true, stdOutSink},
```

Comment:
Since all the tests run in single JVM, the namespaces will be disabled on server side for all of them. Please refactor into multiple classes accordingly.